work_queue/src/work_queue_process.c

```diff
@@ -256,6 +259,81 @@ pid_t work_queue_process_execute(struct work_queue_process *p )
 		if(result == -1)
 			fatal("could not dup pipe to stderr: %s", strerror(errno));
 
+		if(strcmp(p->task->command_line, "@FUNCTION") == 0) {	
+			// read the contents from the input file
+			FILE *fp;
+			char *file_buf = NULL;
+			long numbytes = 0;
+
+			struct work_queue_file *f;
+			list_first_item(p->task->input_files);
+			while((f = list_next_item(p->task->input_files))) {
+				char *file_name = f->remote_name;
+				// printf("file: %s\n", file_name);
+
+				fp = fopen(file_name, "r");
+				
+				if(!fp) {
+					fatal("could not open file: %s", strerror(errno));
+				}
+				
+				// get the file size for reading
+				fseek(fp, 0L, SEEK_END);
+				numbytes = ftell(fp);
+				fseek(fp, 0L, SEEK_SET);
+				
+				// allocate memory and read
+				file_buf = (char *)calloc(numbytes, sizeof(char));
+				fread(file_buf, sizeof(char), numbytes, fp);
```

Comment:
Use `full_fread` from `dttools/full_io` which will ensure that you get all the requested bytes.